### PR TITLE
Add type annotations to `aiven.client.speller` and `tests/test_speller.py`

### PR DESCRIPTION
**Pull request by mlatief**
Add type annotations to:

- [x] `aiven/client/speller.py`
- [x] `tests/test_speller.py`

Part of https://github.com/aiven/aiven-client/issues/258

### CODE REVIEW

File: aiven/client/speller.py [hunk 1]
```diff
@@ -4,21 +4,24 @@
...
+
+
+def suggest(word_to_check: str, known_words: Iterable[str]) -> Optional[str]:
+    def get_correction(word: str) -> Optional[str] :
         """Most probable spelling correction for word."""
         candidates = get_candidates(word)
         return next(iter(candidates)) if candidates else None
 
-    def get_candidates(word):
+    def get_candidates(word: str) -> Iterable[str]:
```
Comment (dmitrii-vasilev) [inline aiven/client/speller.py:16]: Probably can be `Iterator[str]` instead. The result is only used once in `get_correction()`

File: aiven/client/speller.py [hunk 2]
```diff
@@ -4,21 +4,24 @@
...
         """Generate possible spelling corrections for word."""
         return get_known([word]) or get_known(get_edits1(word)) or get_known(get_edits2(word)) or []
 
-    def get_known(words):
+    def get_known(words: Iterable[str]) -> Iterable[str]:
```
Comment (dmitrii-vasilev) [inline aiven/client/speller.py:20]: Also, probably both types can be `Iterator[str]` instead

File: aiven/client/speller.py [hunk 3]
```diff
@@ -4,21 +4,24 @@
...
         """The subset of `words` that appear in the dictionary of WORDS."""
         return set(w for w in words if w in known_words)
 
-    def get_edits1(word):
+    def get_edits1(word: str) -> Iterable[str]:
```
Comment (dmitrii-vasilev) [inline aiven/client/speller.py:24]: Consider `Iterator[str]`

File: aiven/client/speller.py [hunk 4]
```diff
@@ -28,7 +31,7 @@ def get_edits1(word):
         inserts = [L + c + R for L, R in splits for c in letters]
         return set(deletes + transposes + replaces + inserts)
 
-    def get_edits2(word):
+    def get_edits2(word: str) -> Iterable[str]:
```
Comment (dmitrii-vasilev) [inline aiven/client/speller.py:34]: Here it should be `Iterator[str]` because it returns iterator comprehension.

File: tests/test_speller.py [hunk 5]
```diff
@@ -37,5 +38,5 @@
         ("asdf", SERVICE_TYPES, None),
     ],
 )
-def test_suggest(word_to_check, known_words, suggestion):
+def test_suggest(word_to_check: str, known_words: List[str], suggestion: Optional[str]) -> None:
```
Comment (dmitrii-vasilev) [inline tests/test_speller.py:41]: Probably `known_words: Container[str]`

**Review by dmitrii-vasilev — CHANGES_REQUESTED** [review]
I think we can tighten some types here.

File: aiven/client/speller.py (hunk 2 above)
Comment (mlatief) [reply aiven/client/speller.py:20]: I think considering the return value I will keep this one as `Iterable`.

File: aiven/client/speller.py (hunk 3 above)
Comment (mlatief) [reply aiven/client/speller.py:24]: Also here considering the return value I will keep it as `Iterable`.

**Review by dmitrii-vasilev — APPROVED** [review]
LGTM

**Pull request merged**